hubblestack_nova/modules/cve_scan_v2.py

```diff
@@ -0,0 +1,283 @@
+from distutils.version import LooseVersion
+from datetime import datetime
+from time import time as current_time
+
+log = logging.Logger(__name__)
+
+def __virtual__():
+    return True
+
+def audit(data_list, tags, verbose=False):
+    global cache_path
+    os_name = __grains__['os'].lower() 
+    cache_path = '/var/cache/salt/minion/files/base/cve/%s.json' % (os_name)
+    cache = {}
+    #Make cache directory and all parent directories
+    # if it doesn't exist.
+    if not os.path.exists(os.path.dirname(cache_path)):
+        os.makedirs(os.path.dirname(cache_path))
+  
+    # Go through yaml to check for cve_scan_v2,
+    #    if its present, check for a cached version 
+    #    of the scan.
+    for data in data_list:
+
+        if 'cve_scan_v2' in data:
+
+            ttl = data['cve_scan_v2']['ttl']
+            url = data['cve_scan_v2']['url']
+            # Requests only handles http:// requests   
+            if url.startswith('https'):
+                url.replace('https', 'http', 1)
+            cache = _get_cache(ttl, url)
+            if cache.get('result', None) == 'OK':
+                master_json = cache
+                break
+
+    # Query the api.
+    if not cache:
+
+        is_next_page = True
+        page_num = 0
+        
+        # Hit the api, incrementing the page offset until 
+        #   we get all the results together in one dictionary.
+        try:
+            while is_next_page:
+                
+                offset = page_num * 20
+                page_num += 1 
+                url_final = '%s?query=type:%s&order:last year&skip=%s' % (url,os_name, offset)
+                cve_query = requests.get(url_final)
+                cve_json = json.loads(cve_query.text)
+                
+                # Default number of searches per page is 20 so 
+                #    if we have less than that we know this is 
+                #    our last page.
+                if len(cve_json['data']['search']) < 20:
+                    is_next_page = False
+
+                # First page is beginning of master_json that we build on
+                if page_num == 1:
+                    master_json = cve_json
+                    continue
+                
+                master_json = _build_json(master_json, cve_json)
+        
+        except Exception as exc: # Ask about error handling...
+            return
+
+        #Cache results.
+        try:
+            with open(cache_path, 'w') as cache_file:
+                json.dump(master_json, cache_file) 
+        except IOError as exc:
+            log.error('The results weren\'t able to be cached')
+                    
+    ret = {'Success':[], 'Failure':[]}   
+    
+    affected_pkgs = _get_cve_vulnerabilities(master_json)
+    local_pkgs = __salt__['pkg.list_pkgs'](versions_as_list=True)
+    
+    for pkg_obj in affected_pkgs:
+##TODO: for efficiency, should switch the loop to go through just the local_pkgs and check if it's in the affected packages.. 
```

Comment:
can we indent this to the same level as the code? It interrupts the flow of the code to have no indents on this.